cmd/collectors/fabricpool.go

```diff
@@ -0,0 +1,140 @@
+	cache.UUID += ".FabricPool"
+
+	// create flexgroup instance cache
+	for _, i := range data.GetInstances() {
+		if !i.IsExportable() {
+			continue
+		}
+		if match := re.FindStringSubmatch(i.GetLabel("volume")); len(match) == 3 {
+			key := i.GetLabel("svm") + "." + match[1]
+			if cache.GetInstance(key) == nil {
+				fg, _ := cache.NewInstance(key)
+				fg.SetLabels(maps.Clone(i.GetLabels()))
+				fg.SetLabel("volume", match[1])
+			}
+			i.SetExportable(includeConstituents)
+		}
+	}
+
+	l.Logger.Debug().Msgf("extracted %d flexgroup volumes", len(cache.GetInstances()))
+
+	// create summary
+	for _, i := range data.GetInstances() {
+		if match := re.FindStringSubmatch(i.GetLabel("volume")); len(match) == 3 {
+			// instance key is svm.flexgroup-volume
+			key := i.GetLabel("svm") + "." + match[1]
+
+			fg := cache.GetInstance(key)
+			if fg == nil {
+				l.Logger.Error().Str("key", key).Msgf("instance not in local cache")
+				continue
+			}
+
+			for mkey, m := range data.GetMetrics() {
+				if !m.IsExportable() && m.GetType() != "float64" {
+					continue
+				}
+
+				fgm := cache.GetMetric(mkey)
+				if fgm == nil {
+					l.Logger.Error().Str("key", mkey).Msgf("metric not in local cache")
+					continue
+				}
+
+				if value, ok := m.GetValueFloat64(i); ok {
+					fgv, _ := fgm.GetValueFloat64(fg)
+
+					// non-latency metrics: simple sum
+					if !strings.HasPrefix(mkey, "cloud_bin_op_latency_average") {
+						err := fgm.SetValueFloat64(fg, fgv+value)
+						if err != nil {
+							l.Logger.Error().Err(err).Msg("error")
+						}
+						continue
+					}
+
+					// latency metric: weighted sum
+					opsKey := strings.Replace(mkey, "cloud_bin_op_latency_average", opName, 1)
+
+					if ops := data.GetMetric(opsKey); ops != nil {
+						if opsValue, ok := ops.GetValueFloat64(i); ok {
+							var tempOpsV float64
+
+							prod := value * opsValue
+							tempOpsKey := opsKeyPrefix + opsKey
+							tempOps := cache.GetMetric(tempOpsKey)
+
+							if tempOps == nil {
+								if tempOps, err = cache.NewMetricFloat64(tempOpsKey); err != nil {
+									return nil, err
+								}
+								tempOps.SetExportable(false)
+							} else {
+								tempOpsV, _ = tempOps.GetValueFloat64(fg)
+							}
+							if value != 0 {
+								err = tempOps.SetValueFloat64(fg, tempOpsV+opsValue)
+								if err != nil {
+									l.Logger.Error().Err(err).Msg("error")
+								}
+							}
+							err = fgm.SetValueFloat64(fg, fgv+prod)
+							if err != nil {
+								l.Logger.Error().Err(err).Msg("error")
+							}
+						} else {
+							l.Logger.Trace().Msg("no ops value SKIP")
+						}
+					}
+				}
+			}
+		}
+	}
+
+	// normalize latency values
+	for _, i := range cache.GetInstances() {
+		if !i.IsExportable() {
+			continue
+		}
+		for mkey, m := range cache.GetMetrics() {
+			if m.IsExportable() && strings.HasPrefix(mkey, "cloud_bin_op_latency_average") {
```

Comment:
This is code is being used by both zapi and rest plugins, which have different metric names, but we should be able to determine the correct metric name higher in this method and directly get the metric instead of looping over all metrics.
`cache.GetMetrics()[cloudBinOpLatencyAverage]`

Reply:
right, buckets